src/coreclr/runtime.proj

```diff
@@ -78,16 +78,24 @@
     <ItemGroup Condition="('$(TargetsAndroid)' == 'true' or '$(TargetsLinuxBionic)' == 'true') and '$(ANDROID_NDK_ROOT)' != ''">
       <_CoreClrBuildArg Include="-DCMAKE_TOOLCHAIN_FILE=$(ANDROID_NDK_ROOT)/build/cmake/android.toolchain.cmake"/>
       <_CoreClrBuildArg Include="-DANDROID_NDK=$(ANDROID_NDK_ROOT)"/>
-      <_CoreClrBuildArg Include="-DANDROID_STL=none"/>
-      <_CoreClrBuildArg Include="-DANDROID_CPP_FEATURES=&quot;no-rtti no-exceptions&quot;"/>
+      <_CoreClrBuildArg Include="-DANDROID_STL=c++_static"/>
+      <_CoreClrBuildArg Include="-DANDROID_CPP_FEATURES=&quot;no-rtti exceptions&quot;"/>
       <_CoreClrBuildArg Include="-DANDROID_PLATFORM=android-$(AndroidApiLevelMin)"/>
+      <_CoreClrBuildArg Include="-DANDROID_NATIVE_API_LEVEL=$(AndroidApiLevelMin)"/>
       <_CoreClrBuildArg Condition="'$(Platform)' == 'arm64'" Include="-DANDROID_ABI=arm64-v8a" />
       <_CoreClrBuildArg Condition="'$(Platform)' == 'arm'" Include="-DANDROID_ABI=armeabi-v7a" />
       <_CoreClrBuildArg Condition="'$(Platform)' == 'x86'" Include="-DANDROID_ABI=x86" />
       <_CoreClrBuildArg Condition="'$(Platform)' == 'x64'" Include="-DANDROID_ABI=x86_64" />
 
-      <!-- No LTTNG on Android -->
-      <_CoreClrBuildArg Include="-cmakeargs -DFEATURE_EVENT_TRACE=0"/>
+      <!-- 
+        TODO: No LTTNG on Android, it's disabled at the eventprovider level. Will need an answer. 
+        Disabling FEATURE_EVENT_TRACE is too wide.
+      -->
+      <!--
+        Unsure if something like this is actually defined. We need this to differentiate between
+        features that are enabled in mono android, but not in coreclr.
+      -->
+      <_CoreClrBuildArg Include="-cmakeargs -DRUNTIME_CORECLR=1" />
```

Comment:
Is this used anywhere?

Reply:
Yeah, one spot. Is there a better existing way to do this?

https://github.com/dotnet/runtime/blob/9779aeba4cb44df9159122dae10126f3293acb54/src/native/eventpipe/CMakeLists.txt#L17

Reply:
Could we add a general feature flag that the Mono vs CoreCLR builds would set?

Is this expected to work for the NativeAOT build?

Reply:
Added a flag in the `eng/native/configuretools.cmake` as it seems to be shared.

Reply:
Fixed a typo in https://github.com/dotnet/runtime/pull/110471/commits/a5d1115e4eb9f3b7dfe3e1534c97ed647bd68bf6